### ISSUE TITLE
Update Test infrastructure: https://mylyn.org does not have a valid certificate

### ISSUE THREAD

**Issue by BeckerFrank** [eclipse-mylyn/.github#3]
In https://github.com/eclipse-mylyn/org.eclipse.mylyn.tasks/pull/5# we have 7 failed tests.

These 7 are related to ssl PKIX errors because the URL used https://mylyn.org does not have a valid certificate.
I send an email and ask Tasktop if they can renew the old certificate that is no longer valid. 
I also ask what will happen to the domain in the future.

**Comment by ruspl-afed** [comment]
Thanks for submitting @BeckerFrank , I've changed title and description to be more specific

**Comment by wimjongman** [comment]
Lovely!

**Comment by gnl42** [comment]
I un-commented the 'removed' tests in AllHudsonsTest in my Jenkins authorization branch and all the tests pass so I guess the Hudson/Jenkins server are online at moment.

@BeckerFrank  Should I commit the updated AllHudsonsTest  or are the servers availability suspect?

**Comment by BeckerFrank** [comment]
> I un-commented the 'removed' tests in AllHudsonsTest in my Jenkins authorization branch and all the tests pass so I guess the Hudson/Jenkins server are online at moment.
> 
> @BeckerFrank Should I commit the updated AllHudsonsTest or are the servers availability suspect?

Yes, we can activate the tests. But we should build a new Test Server because we do not know how long mylyn.org will be there. https://www.eclipse.org/lists//mylyn-dev/msg02496.html

**Comment by gnl42** [comment]
I had to revert the revert. test servers too unstable

**Comment by BeckerFrank** [comment]
> I had to revert the revert. test servers too unstable

What do you mean with unstable?
What test instance did you use?

**Comment by gnl42** [comment]
They all pass when I run them in Eclipse, but fail when run by Maven:
>
> Running repository: https://mylyn.org/hudson-3.3.3/ 
> ***@***.***(org.eclipse.mylyn.commons.sdk.util.RepositoryTestFixture$Activation)
>
> Running 
> testAbortBuild(org.eclipse.mylyn.hudson.tests.client.HudsonClientTest)
>
> [ERROR]
>
> Running 
> testGetJobsWithWhitespaces(org.eclipse.mylyn.hudson.tests.client.HudsonClientTest)
>
> [ERROR]
>
> Running 
> testRunBuildSucceeding(org.eclipse.mylyn.hudson.tests.client.HudsonClientTest)
>
> [ERROR]
>
> Running 
> testGetJobs(org.eclipse.mylyn.hudson.tests.client.HudsonClientTest)
>
> [ERROR]
>
> Running 
> testGetJobDisabled(org.eclipse.mylyn.hudson.tests.client.HudsonClientTest)
>
> Running 
> testGetNestedJobs(org.eclipse.mylyn.hudson.tests.client.HudsonClientTest)
>
> Running 
> testValidateExpiredCookie(org.eclipse.mylyn.hudson.tests.client.HudsonClientTest)
>
> [ERROR]
>
> Running 
> testRunBuildFailing(org.eclipse.mylyn.hudson.tests.client.HudsonClientTest)
>
> [ERROR]
>
> Running 
> testValidateValidUrl(org.eclipse.mylyn.hudson.tests.client.HudsonClientTest)
>
> Running 
> testGetJobGit(org.eclipse.mylyn.hudson.tests.client.HudsonClientTest)
>
> [ERROR]
>
> Running 
> testGetManyJobs(org.eclipse.mylyn.hudson.tests.client.HudsonClientTest)
>
> [ERROR]
>
> Running 
> testRunNestedJob(org.eclipse.mylyn.hudson.tests.client.HudsonClientTest)
>
> Running 
> testValidateValidUrlAuthenticate(org.eclipse.mylyn.hudson.tests.client.HudsonClientTest)
>
> [ERROR]
>
> Running 
> testRunBuildGuest(org.eclipse.mylyn.hudson.tests.client.HudsonClientTest)
>
> Cancelling nested steps due to timeout
>
> Sending interrupt signal to process
>

Not sure what you mean by instance? I'm using whatever the code is set 
up for by default.

**Comment by BeckerFrank** [comment]
Yes, by instances I mean the services you can find here https://mylyn.org and these instances are used for the unit tests.

Are you sure that al the errors you get are not because we have an problem within the java client code?

**Comment by gnl42** [comment]
Pretty sure. The only thing I can think of is that Eclipse provides 
something in the runtime environment that is missing with Maven.

**Comment by BeckerFrank** [comment]
Can you debug on of the failing tests, so that we have more details?

**Comment by gnl42** [comment]
I had no luck debugging the maven run.
Maven seems to ignore -Dtest=HudsonAllTests plus the unit tests are run in a new java process so I can't debug it with M2e and I had no luck attaching a debugger to the process.

I did discover that if I set surefire:useUIHarness=false all but one of the tests pass.
I commented out the failing test (HudsonClientTest.testRunBuildGuest() (why doesn't @Ignore work?)). and committed the changes.

Baring some accidental insight I've gone as far as I can at the moment with this.
